articleview.cc

```diff
@@ -2669,6 +2661,27 @@ void ArticleView::highlightFTSResults()
   ftsSearchIsOpened = true;
 }
 
+void ArticleView::highlightAllFtsOccurences( QWebPage::FindFlags flags )
+{
+  flags |= QWebPage::HighlightAllOccurrences;
+
+  // Usually allMatches contains mostly duplicates. Thus searching for each element of
+  // allMatches to highlight them takes a long time => collect unique elements into a
+  // set and search for them instead.
+  // Don't use QList::toSet() or QSet's range constructor because they reserve space
+  // for QList::size() elements, whereas the final QSet size is likely 1 or 2.
+  QSet< QString > uniqueMatches;
+  for( int x = 0; x < allMatches.size(); ++x )
+  {
+    QString const & match = allMatches.at( x );
+    // Consider words that differ only in case equal if the search is case-insensitive.
+    uniqueMatches.insert( ftsSearchMatchCase ? match : match.toLower() );
+  }
+
+  for( QSet< QString >::const_iterator it = uniqueMatches.constBegin(); it != uniqueMatches.constEnd(); ++it )
+    ui.definition->findText( *it, flags );
+}
```

Comment:
findText in webengine , only the last findText take effect ,
in webkit engine ,  all the findText can take effect at the same time?

Reply:
Yes. From https://doc.qt.io/archives/qt-5.5/qwebpage.html#findText:

If the [HighlightAllOccurrences](https://doc.qt.io/archives/qt-5.5/qwebpage.html#FindFlag-enum) flag is passed, the function will highlight all occurrences that exist in the page. All subsequent calls will extend the highlight, rather than replace it, with occurrences of the new string.